doc/types/NodeDefinition.adoc

```diff
@@ -0,0 +1,205 @@
+
+| `sounds` | `{type-nodesoundspec}` | Definition of node sounds to be played at various events. See {type-nodesoundspec}. | `nil`
+
+| `drop` | `{type-string} or {type-nodedropspec}` | By default, the dropped item is the node itself. If a {type-string} is given, `drop` will be dropped (for example: `"default:stone"`). Using a table allows multiple items, drop chances and item filtering, see {type-nodedropspec}. | `""`
+
+| `mod_origin` | `{type-string}` | stores which mod actually registered a node. If the source could not be determined, it contains `"??"`. This is useful for getting which mod truly registered something.
+
+Example: if a node is registered as `":othermodname:nodename"`, nodename will show `"othermodname"`, but mod_origin will say `"modname"`. | `"<modname>"`
+
+| `on_construct` | `{type-function}(pos:{class-vector})` | The node constructor function called after adding the node to the world. This can, for example, be used to add metadata to the node.
+
+NOTE: This function is not called for bulk node placement (i.e. schematics and VoxelManip).
+
+WARNING: Within an on_construct callback, minetest.set_node can cause an infinite loop if it invokes the same callback. Consider using `minetest.swap_node` instead. | `nil`
+
+| `on_destruct` | `{type-function}(pos:{class-vector})` | The node destructor function called before removing the node from the world.
+
+NOTE: Like in `on_construct`, this function is not called for bulk node placement. | `nil`
+
+| `after_destruct` | `{type-function}(pos:{class-vector}, oldnode:{class-noderef})` | he node destructor function called after removing the node from the world.
+
+NOTE: Like in `on_construct`, this function is not called for bulk node placement. | `nil`
+
+| `after_place_node` | `{type-function}(pos:{class-vector}, placer:{class-playerref}, itemstack:{class-itemstack}, pointed_thing:{type-pointedthing})` | Called after constructing the node when the node was placed using `minetest.item_place_node` or `minetest.place_node`. If the function returns true, no item is taken from the {class-itemstack}. `placer` may be any valid {class-objectref} or {type-nil}. | `nil`
+
+| `after_dig_node` | `{type-function}(pos:{class-vector}, oldnode:{class-noderef}, oldmetadata:{type-table}, digger:{class-playerref})` | Called after destructing the node when the node was dug using `minetest.node_dig` or `minetest.dig_node`. | `nil`
+
+| `can_dig` | `{type-function}(pos:{class-vector}, player:{class-playerref}\|{type-nil})` | If the node shall be diggable, this functions should return true, false otherwise. | `nil`
+
+| `on_dig` | `{type-function}(pos:{class-vector}, node:{class-noderef}, pointed_thing:{type-pointedthing})` | Called when the node was dug. By default, this checks privileges (if the node can be dug). If the item used to dig this node is a tool, the tools durability will be deducted. Return true if the node was dug successfully, false otherwise. | `nil`
```

Comment:
Default is not `nil` but rather `minetest.node_dig`. I feel it would make sense to move this above the corresponding `after_*` callback to make this clear.